clients/src/main/java/org/apache/kafka/common/requests/LeaderAndIsrRequest.java

```diff
@@ -145,24 +145,21 @@ public LeaderAndIsrResponse getErrorResponse(int throttleTimeMs, Throwable e) {
                         .setErrorCode(error.code()));
             }
             responseData.setPartitionErrors(partitions);
-            return new LeaderAndIsrResponse(responseData, version());
-        }
-
-        List<LeaderAndIsrTopicError> topics = new ArrayList<>(data.topicStates().size());
-        Map<String, Uuid> topicIds = topicIds();
-        for (LeaderAndIsrTopicState topicState : data.topicStates()) {
-            LeaderAndIsrTopicError topicError = new LeaderAndIsrTopicError();
-            topicError.setTopicId(topicIds.get(topicState.topicName()));
-            List<LeaderAndIsrPartitionError> partitions = new ArrayList<>(topicState.partitionStates().size());
-            for (LeaderAndIsrPartitionState partition : topicState.partitionStates()) {
-                partitions.add(new LeaderAndIsrPartitionError()
+        } else {
+            for (LeaderAndIsrTopicState topicState : data.topicStates()) {
+                List<LeaderAndIsrPartitionError> partitions = new ArrayList<>(
+                    topicState.partitionStates().size());
+                for (LeaderAndIsrPartitionState partition : topicState.partitionStates()) {
+                    partitions.add(new LeaderAndIsrPartitionError()
                         .setPartitionIndex(partition.partitionIndex())
```

Comment:
Could you add comments to explain why we don't need to set topic name here? Otherwise, It looks like a bug to me :(

Reply:
@chia7712 Starting from version 5, the topic name is no longer in the response: https://github.com/apache/kafka/blob/trunk/clients/src/main/resources/common/message/LeaderAndIsrResponse.json#L46 so we don't set it anymore. Do you really want to add a comment for this? That seems unnecessary to me.

Reply:
> Do you really want to add a comment for this? That seems unnecessary to me.

I don't have strong reason for that. As you mentioned, we can read protocol file to understand the reasons of ignoring that filed.

Reply:
Ack. So let's keep it as is.